packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -1883,6 +1884,53 @@ abstract class MultiSelectableSelectionContainerDelegate extends SelectionContai
     selectable.removeListener(_handleSelectableGeometryChange);
   }
 
+  @override
+  int? get contentLength {
+    int length = 0;
+    for (final Selectable selectable in selectables) {
+      length += selectable.contentLength ?? 0;
+    }
+    return length;
+  }
+
+  @override
+  TextSelection? get textSelection {
```

Comment:
Can this calculation go in the override of `getSelectedContent`? This way the `SelectedContent` have the most up to date `TextSelection` and `_updateSelectedContentIfNeeded` will only need to call `getSelectedContent()` rather than having to access `_selectable.textSelection`. Let me know if this is not possible, or if I can provide a better explanation.